mtags-interfaces/src/main/java/scala/meta/pc/CodeActionId.java

```diff
@@ -0,0 +1,10 @@
+package scala.meta.pc;
+
+public enum CodeActionId {
```

Comment:
With an existing enum we always have to add a new id to mtags-interfaces.

Maybe we should have something like:
```
interface CodeAction{
  String id;
  CodeActionData data;
}

interface CodeActionData{

}
```

This way we can implement things in metals or even within compiler itself independently. :thinking:

Reply:
Ok, makes sense. But if we go with `String` ids can we just use them explicitly as the method parameters?

Or do you mean that when creating a new code action one should create a new object that implements `CodeAction`? This would still have to be added to the common java sources, right? Or should it just use anonymous instances of `CodeAction`?

The most source compat way is probably to use `String` ids, and `Object` type payload. This way we don't need to define new classes in mtags-interfaces for every new code action.

Maybe the result of the `codeAction` method should also return `Object` in that case?

Reply:
Ach right, for this to make sense we would need to figure out some kind of basic interface and I don't see any common parts here :thinking: 

I think we might need to anyway have to add those class to common java sources. The only alternative is to send a gson object, which would not be fun to work with.

Reply:
So maybe keeping a `String` id and an `Object` payload might make the most sense?
In that case, we would be able to have payloads with existing classes (like lists of names) for free and when needed, we would add new classes to the common java sources and use those.

Reply:
Sure, not perfect, but makes sense!